src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -671,6 +676,14 @@ public function merge(self $container)
 
             $this->autoconfiguredInstanceof[$interface] = $childDefinition;
         }
+
+        foreach ($container->getAutoconfiguredAttributes() as $attribute => $configurator) {
+            if (isset($this->autoconfiguredAttributes[$attribute])) {
+                throw new InvalidArgumentException(sprintf('"%s" has already been autoconfigured and merge() does not support merging autoconfiguration for the same attribute.', $attribute));
```

Comment:
This is an issue. It means one cannot replace the rules if they want to. I think we should override instead.
Or maybe this is solved by using instance-of-conditionals?

Reply:
I'm mimicking the behavior of `getAutoconfiguredInstanceof()` rules (see the `foreach` block directly above this one). I'm unsure how predictable a mechanism to override existing rules is for the developer.

If an app wants to override behavior, maybe it's better that app defines its own attribute for its own rules?

Reply:
OK, let's keep it as is then!